src/main/minio.js

```diff
@@ -244,20 +247,22 @@ export class Client {
     this.userAgent = `${this.userAgent} ${appName}/${appVersion}`
   }
 
-  // partSize will be atleast minimumPartSize or a multiple of minimumPartSize
-  // for size <= 50GiB partSize is always 5MiB (10000*5MiB = 50GiB)
-  // for size > 50GiB partSize will be a multiple of 5MiB
-  // for size = 5TiB partSize will be 525MiB
+  // Calculate part size given the object size. Part size will be atleast this.partSize
   calculatePartSize(size) {
     if (!isNumber(size)) {
       throw new TypeError('size should be of type "number"')
     }
     if (size > this.maxObjectSize) {
       throw new TypeError(`size should not be more than ${this.maxObjectSize}`)
     }
-    var partSize = Math.ceil(size/10000)
-    partSize = Math.ceil(partSize/this.minimumPartSize) * this.minimumPartSize
-    return partSize
+    var partSize = this.partSize
+    while (true) {
+      if ((partSize * 10000) > size) {
```

Comment:
Why was the old code removed? It looks good to me. Even if the user provides a `partsize`, will we override that value here?

Reply:
> Even if the user provides a partsize, will we override that value here?

No, if user overrides, then that value is used for the partSize

Reply:
ok